SSL-enabled Nginx Ingress Controller integrated Hashicorp Vault

devlinmr opened the pull request:
Based on the nginx-alpha controller, this adds functionality to configure SSL nginx servers for ingresses requiring it, pulling in certificates from Vault.

See README.md.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/553)

<!-- Reviewable:end -->

aledbf committed:
Add support for private repositories

aledbf committed:
Use git credential cache instead of .netrc file. Remove default change of permission

aledbf committed:
Allow running git as nobody

aledbf committed:
Update nginx-slim

aledbf committed:
Update haproxy image

aledbf committed:
Update keepalived image

aledbf committed:
Use ubuntu-slim image

aledbf committed:
Use ubuntu-slim image

Commit:
Adds support for no_proxy

Previously, the Ansible project supported setting http/https proxy
settings, but not no_proxy. Bypassing the http/https proxy is
common in production deployments. For example, a deployment may
need to bypass the http/https proxy to pull images from a local
Docker registry.

aledbf committed:
Add signal handlers

aledbf committed:
Change docker images location

Commit:
Vault-enabled SSL ingress controller based on nginx-alpha

Commit:
Vault-enabled SSL ingress controller based on nginx-alpha

Commit:
Readme

Commit:
Readme

Commit:
Retouch README

Commit:
Add RC example, remove Makefile

Commit:
Add RC example, remove Makefile

Commit:
Godeps

Commit:
Godeps

Commit:
Godeps

Commit:
Godeps

Commit:
Mofify Godeps again

Commit:
Mofify Godeps again

Commit:
More Gedeps

Commit:
More Gedeps

Commit:
Godeps again

Commit:
Fix issue with restarts under load and logging to stdout with -s reload

Commit:
Fix issue with restarts under load and logging to stdout with -s reload

bprashanth commented:
Hi @devlinmr thanks for the PR. Didn't notice it till now, will take a look by monday. Feel free to tag me or @kubernetes/goog-cluster on Ingress/lb prs in the future.

bprashanth commented on ingress/controllers/nginx-alpha-ssl/README.md:
```diff
@@ -0,0 +1,129 @@
+
+/etc/vault-token/ingress-read-only
+
+An example for creating this as a kubernetes secret is shown below.
```
did you forget to add this documentation, or did I miss it?

bprashanth replied:
Ah, I see a subsequent commit has it

bprashanth commented on ingress/controllers/nginx-alpha-ssl/controller.go:
```diff
@@ -0,0 +1,217 @@
+		vaultEnabled = false
+	}
+
+	token := os.Getenv("VAULT_TOKEN")
```
How about actually adding an RC+Service yaml for vault, and then polling here till the vault service is available (--wait-for-value or something can turn off the polling, but if it's true we just don't proceed until vault is up).

bprashanth commented on ingress/controllers/nginx-alpha-ssl/controller.go:
```diff
@@ -0,0 +1,217 @@
+				ingressHost := ingress.Spec.Rules[0].Host
+				vaultPath := "secret/ssl/" + ingressHost
+
+				keySecretData, err := vault.Logical().Read(vaultPath)
```
can you include some docs on how one inserts certs into value? for bonus points you can make a simple script that is similar to `kubectl create -f secret.yaml` but inserts into vault instead of apiserver.

bprashanth commented on ingress/controllers/nginx-alpha-ssl/run.sh:
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+export VAULT_SKIP_VERIFY=true
+[ -f /etc/vault-token/ingress-read-only ] && export VAULT_TOKEN=`cat /etc/vault-token/ingress-read-only`
```
Why not just read this directly in the controller (--vault-token) instead of using an env var? 
If you add it as a secret it will be mounted in the specified location (or it's a bug in Kubernetes).

bprashanth commented:
Did a quick scan. This is nice! If you could add a section about "why use valut?" (just so others can make an informed decision about weather or not they want to deploy the extra rc+svc), add vault as a svc+rc, and squash the godep commits I'll take a second look.

bprashanth commented:
I think travis is complaining about missing Godeps

Commit:
Ansible: Add flannel support for Ubuntu 14.04

This changes adds flannel support for Ubuntu 14.04. It follows the same
pattern of etcd playbook to use flannel release from Github, and
service install with upstart.

aledbf committed:
Improve event handling using a workqueue

rutsky committed:
include proper flannel_opts on VBox provider

It's ugly due to the fact that there is no simple way to get current
provider, see <https://github.com/mitchellh/vagrant/issues/1867> for
discussion.

I made some variables global (with "$" sign), since they are more like
global constants then local variables, plus in this ways it's easier
to use them inside functions.

rutsky committed:
set proper public IP on Virtualbox provider in vagrant

On VirtualBox eth0 is used for NAT-ed internet connection, and actual
connectivity with the host and other VMs is done using eth1 interface.

eparis committed:
New munger to manage 'release-note' labels

Rules:
All PRs must have one of
- `release-note`
- `release-note-action-required`
- `release-note-none`
- `release-note-label-needed`
Add `release-note-label-needed` to any PR without one of the above.

If a PR gets lgtm and still has `release-note-label-needed` remove LGTM and
message where to get more info.

gitschaub committed:
Ansibilizes CoreOS bootstrapping process. Removes check for .bootstrapped

eparis committed:
Merge pull request #604 from danehans/ansible_dist_fix

Fixes Ansible fact for determining CoreOS

Commit:
Updates the Default bin_dir for Ansible

Previously, the default bin_dir did not work for some platforms.
The new default follows the contrib/init project settings.

Fixes Issue: https://github.com/kubernetes/contrib/pull/618

Commit:
Fixes python_bin path for Kube Addons in Ansible

Previously, the python_path was not being properly updated for
CoreOS systems. This patch moves python_bin fact management
from the kubernetes-addons role to the kubernetes role.

Fixes Issue: https://github.com/kubernetes/contrib/issues/623

eparis committed:
Merge pull request #624 from danehans/addons_fix

Fixes python_bin path for Kube Addons in Ansible

eparis committed:
Merge pull request #622 from danehans/bin_def

Updates the Default bin_dir for Ansible

rutsky committed:
use new "public_iface" variable in vagrant-ansible.yml

This variable specifies interface that should be used as public
interface.
Useful on providers such as VirtualBox, where eth0 is used for
NAT-ed internet connection, and actual connectivity with the
host and other VMs is done using eth1 interface.

Suggested by @atosatto.

rutsky committed:
use ansible_default_ipv4.interface instead of "eth0" for default interface

According to @eparis suggestion.

gmarek committed:
Add retries to reading data from GCS as it's not reliable enough.

eparis committed:
Merge pull request #631 from gmarek/munger

Add retries to reading data from GCS as it's not reliable enough.

eparis committed:
Merge pull request #619 from rutsky/ansible-vagrant-centos7

fix ansible+vagrant+virtualbox+centos7 configuration

rutsky committed:
specify OS image using env variable OS_IMAGE + add Core OS image

rutsky committed:
document OS_IMAGE env variable + add other notes about running Vagrant

rutsky committed:
apply pre-absible role invagrant-ansible.yml

Allows to run invagrant-ansible.yml on Core OS.

rutsky committed:
create /etc/hosts if it is not present in vagrant-ansible.yml

On stock Core OS there is no /etc/hosts file.

eparis committed:
Include the title and first comment in submit queue merges

This allows us to find that information in the git logs for later
processing.

eparis committed:
mungegithub: Remove cp-candidate label if the words "Automated cherry pick of #NNNNN" are in the branch

Previously we looked for messages about the MERGE commit that went to
master (aka using git cherry-pick). But the scripts in hack use `git
am`. Though thankfully, at least for single picks, they have an easy to
figure out distinctive title.

Commit:
Create container for mailing via remote smtp

caesarxuchao committed:
Merge pull request #547 from caesarxuchao/chao

Some improvements to micro-demo

bprashanth committed:
Merge pull request #570 from aledbf/nginx-ingress-use-endpoints

Update nginx controller to use endpoints. Remove lua to configure upstreams

lavalamp committed:
Merge pull request #634 from krousey/shamemailer

Create container for mailing via remote smtp

bprashanth committed:
Merge pull request #552 from aledbf/update-images

Use ubuntu-slim image

Commit:
Updates Vagrant Documentation

Previously, the Vagrant documentation was missing a few minor
steps. This patch adds verbosity to the Vagrant documentation that
should improve developer onboarding.

gosharplite committed:
Update README.md

bprashanth committed:
Merge pull request #493 from aledbf/private-git-sync

Add support for private repositories in git-sync

bprashanth committed:
Merge pull request #639 from gosharplite/patch-1

HTTP -> HTTPS

bprashanth committed:
Fix some typos in nginx README.md

eparis committed:
Merge pull request #637 from danehans/vagrant_docs

Updates Vagrant Documentation

bprashanth committed:
Merge pull request #640 from bprashanth/nginxdoc

Fix some typos in nginx README.md

bprashanth committed:
Simplify urlmap update logic.

bprashanth committed:
Single sync error.

aledbf committed:
Improve documentation. Add flag to enable vts status module

eparis committed:
Merge pull request #644 from rutsky/ansible-vagrant-coreos

add CoreOS support to Ansible+Vagrant+VirtualBox + fix initial vagrant-ansible.yml playbook

Commit:
Changes source_type default to packageManager

eparis committed:
Merge pull request #645 from danehans/pkg_mgr

Changes source_type default to packageManager

eparis committed:
Merge pull request #630 from danehans/no_proxy

Adds support for no_proxy

aledbf committed:
Rename controller to nginx

bprashanth committed:
Clarify firewall rule doc.

bprashanth committed:
Merge pull request #647 from bprashanth/ing_docs

Clarify firewall rule doc.

rutsky committed:
use same /etc/sysconfig/flanneld env file as on other distributions

For example in CentOS 7 flanneld.service contains:

ExecStart=/usr/bin/flanneld -etcd-endpoints=${FLANNEL_ETCD} \
  -etcd-prefix=${FLANNEL_ETCD_KEY} $FLANNEL_OPTIONS

Which allows to use variables like FLANNEL_ETCD in
/etc/sysconfig/flanneld.

Currently used approach on CoreOS uses flannel-specific environment
variables FLANNELD_*. However there is no way to pass arbitrary options
to flanneld daemon using FLANNELD_* environment variables (and currently
used FLANNELD_OPTIONS doesn't work actually).

Lets start flanneld in the same way on CoreOS to unify using of
environment files.

bprashanth committed:
Merge pull request #563 from bprashanth/ingress_static_ip

Allow a user specified global static ip via annotation.

Commit:
Updates Ansible Vagrant Documentation

93aae647510d29426c0de8d1c6bec875941 updated the default
package_manager. This commit updates the Vagrant doc to support
this new default setting.

8417f092090067dbbf8096815d50edaf77bfea2f introduced a Vagrant doc
bug. This commit updates references to Core OS 7 and coreos7 being
the default Vagrant image to fix this bug.

aledbf committed:
Remove examples from the main README.md

bprashanth committed:
List from cloud and resync InMemoryPool.

eparis committed:
Merge pull request #649 from danehans/vagrant_docs

Updates Ansible Vagrant Documentation

eparis committed:
Merge pull request #648 from rutsky/fix-coreos-flannel-options

use same /etc/sysconfig/flanneld env file as on other distributions

gitschaub committed:
Updated adhoc/uninstall.yml to support CoreOS

eparis committed:
Merge pull request #653 from adamschaub/coreos_uninstall

Updated adhoc/uninstall.yml to support CoreOS

Commit:
Fixes Ansible Vagrant for CoreOS

Previously, Vagrant would not properly set the box download
url:

kube-node-1: Downloading: https://storage.googleapis.com/\
.release.core-os.net/amd64-usr//coreos_production_vagrant.json

An error occurred while downloading the remote file. The error
message, if any, is reproduced below. Please fix this error and
try again.

The requested URL returned error: 400 Bad Request

This commit properly sets the coreos update_channel
and image_version variables.

Commit:
Updates Vagrant README

1. Adds Ansible 2.0.0.2 for pip install
2. Updated NUM_MINIONS to NUM_NODES

The default version of Ansible for Mac OSX 10.10.5 causes errors
with coreos Ansible playbooks. THis commit hardcodes an Ansible
version that is compatible acorss distros. The Vagrantfile was
updated in a previous commit to go from NUM_MINIONS > NUM_NODES
but the README was not updated. This commit updates NUM_MINIONS
to NUM_NODES in the Vagrant readme.

gitschaub committed:
Touch up uninstall CoreOS wording. Improvements and parameterization

eparis committed:
Merge pull request #588 from unicell/flannel-on-ubuntu

Ansible: Add flannel support for Ubuntu 14.04

bprashanth committed:
Retry GCE client creation.

ingvagabund committed:
Init the did_install fact in packageManagerInstall.yml for node role.
Otherwise it is undefined for other distros than CoreOS or Fedora

eparis committed:
Merge pull request #656 from danehans/vagrant_coreos

Fixes Ansible Vagrant for CoreOS

rutsky committed:
fix "--v" flag description

It's "verbosity" flag, the higher it is, the more verbose messages will
be printed.

It should mimic glog's "--v":
<https://google-glog.googlecode.com/svn/trunk/doc/glog.html#verbose>

rutsky committed:
download Kubernetes binaries one by one from artifacts server

Kubernetes GitHub release tarball has size of about 400+ MB.
Previously it's being downloaded, extracted, and inner tarball extracted
one more time (on each host).
This took roughly 400+ MB * NUM_NODES traffic and about 1 GB of space in
/tmp/ (on each host).

Starting from 1.2.0 it takes more than 1 GB of space in /tmp by Ansible,
which fails for example on Core OS, where /tmp is 1 GB ramdisk.

With these changes only required binaries are downloaded (not
compressed, but it's still significantly less than whole release tarball size).

rutsky committed:
bump installed in Core OS kubernetes version to 1.2.0

rutsky committed:
download Flannel releases to /opt/

/usr/ is readonly on Core OS, and /opt/ should work on Ubuntu too.

rutsky committed:
move "flannel_releases_dir" to variables defaults + remove trailing slash

eparis committed:
Merge pull request #661 from rutsky/fix-flannel-installation-coreos

download Flannel releases to /opt/

eparis committed:
Merge pull request #662 from rutsky/fix-v-description

fix "--v" flag description

eparis committed:
Merge pull request #632 from eparis/merge-include-title

Include the title and first comment in submit queue merges

eparis committed:
Merge pull request #658 from ingvagabund/set-implicit-value-for-did_install-fact-for-node-role

Set implicit value for did install fact for node role

rutsky committed:
download Kubernetes binaries to versioned directory and link files from it

eparis committed:
Merge pull request #663 from rutsky/download-k8s-binaries-one-by-one

Bump K8s version to 1.2.0 on Core OS + download k8s binaries one by one

Commit:
Add Amey-D and wonderfly to the whitelist.

rutsky committed:
add requirement for minimal Vagrant version to use for Core OS images

Vagrant prior version 1.8.0 doesn't write group variables into inventory file:
<https://github.com/mitchellh/vagrant/commit/dd4ae1a51cfb246d561bced89d3b34ee90a0a38f>

eparis committed:
Use the issue Body not the first commit in merges

eparis committed:
Merge pull request #621 from adamschaub/coreos_python_bootstrap

Ansibilizes CoreOS bootstrapping process.

eparis committed:
Merge pull request #665 from roberthbailey/whitelist

Add Amey-D and wonderfly to the whitelist.

eparis committed:
Merge pull request #668 from eparis/fix-merge-include-title

Use the issue Body not the first commit in merges

gitschaub committed:
CoreOS pre-ansible: un-hardcode bin_dir

eparis committed:
Merge pull request #615 from eparis/release-notes-labeler

New munger to manage 'release-note' labels

eparis committed:
Enable the release note munger

bprashanth committed:
Merge pull request #642 from aledbf/restrict-nginx-status

Improve nginx-ingress-controller documentation.

rutsky committed:
update kube-addons scripts with upstream version

This fixes issues #626 and #667.

Now they kube-addon-update.sh and kube-addons.sh downloaded from master
branch of Kubernetes project.

Also adds restart notification for kube-addons when scripts are updated.

Upstream version of these scripts uses custom Docker image with Python
when system Python is not available.
This image is built if Ansible runs on Core OS.

aledbf committed:
Update nginx to 1.9.13

aledbf committed:
Add ump load balancing and naxsi as WAF. Update nginx to 1.9.13

aledbf committed:
Add configuration for retries in non-idempotent requests

eparis committed:
Merge pull request #669 from adamschaub/coreos_python_bootstrap

CoreOS pre-ansible: un-hardcode bin_dir

luxas committed:
Add cross-compiling support for exechealthz

Commit:
updated kube-ui to use new kubernetes dashboard ui https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/dashboard

merged with upstream master

Update kube-addons.service.j2

Reverted the KUBECTL_BIN to the correct value, since the {{ bin_dir }} was referencing the common.yaml default which is /usr/local/bin
And there is not kubectl overthere

Update kube-addons.upstart.j2

Reverted the KUBECTL_BIN to the correct value, since the {{ bin_dir }} was referencing the common.yaml default which is /usr/local/bin
And there is not kubectl overthere

eparis committed:
Merge pull request #672 from rutsky/update-kube-addons-with-upstream

update kube-addons scripts with upstream version

eparis committed:
Merge pull request #676 from deromka/master

Updated kube-ui to use new kubernetes dashboard ui

bprashanth committed:
Merge pull request #675 from aledbf/add-udp-lb

Add UDP load balancing to nginx ingress controller

aledbf committed:
Update Ingress status information in nginx controller

eparis committed:
Merge pull request #633 from eparis/clear-cp-candidate-when-picked-via-script

Clear cp candidate when picked via hack script

bprashanth committed:
Merge pull request #646 from luxas/exechealth_arm

Add cross-compiling support for exechealthz

eparis committed:
Switch from wheezy to jessie

So we can get a modern version of git

eparis committed:
cherrypick: Do not log every time a PR doesn't mater

This was a debug line which was left in.

eparis committed:
Merge pull request #683 from eparis/less-log

cherrypick: Do not log every time a PR doesn't mater

eparis committed:
cherrypick: Do not use --fork-base in git merge-base

The version of git in wheezy doesn't have fork-base. The version in
jessie doesn't work correctly.

eparis committed:
Merge pull request #684 from eparis/jessie

cherrypick: Do not use --fork-base in PR matching

aledbf committed:
Avoid sync Ingress updates

aledbf committed:
Update terminationGracePeriodSeconds in examples

eparis committed:
cherrypick: auto-approve cherrypicks if the pick to master was approved

bprashanth committed:
Merge pull request #674 from aledbf/update-nginx-1913

Update nginx to 1.9.13

bprashanth committed:
Clarify default backend creation.

gmarek committed:
Add more logging to GCS stable check

bprashanth committed:
Merge pull request #687 from bprashanth/README

Clarify default backend creation.

eparis committed:
Also set the milestone for the non-master PR

eparis committed:
Add a test case

eparis committed:
Merge pull request #688 from gmarek/munger

Add more logging to GCS stable check

eparis committed:
Merge pull request #686 from eparis/auto-approve-cherry-picks

cherrypick: auto-approve cherrypicks if the pick to master was approved

eparis committed:
Merge pull request #670 from eparis/enable-release-note-label

Enable the release note munger

eparis committed:
Enable the cp auto approve munger

eparis committed:
Merge pull request #691 from eparis/enable-cp-auto-approve

Enable the cp auto approve munger

eparis committed:
mungegithub: Fix logo since it moved in the kube repo

eparis committed:
Merge pull request #692 from eparis/logo-location

mungegithub: Fix logo since it moved in the kube repo

eparis committed:
mungegithub: and fix the other bad logo link

eparis committed:
Merge pull request #693 from eparis/really-fix-logo

mungegithub: and fix the other bad logo link

aledbf committed:
Add test to verify SSL certificate creation

ingvagabund committed:
Enable security context in clusters by default
- and make the list of admission controllers configurable

eparis committed:
Merge pull request #697 from ingvagabund/unset-security-context-deny-in-admission-control

Enable security context in clusters by default

gmarek committed:
Add even more logging to munger

eparis committed:
Merge pull request #698 from gmarek/munger

Add even more logging to munger

eparis committed:
submit queue: Update chip links

They used to point to
storage.cloud.google.com/
But that stopped working so point to
console.cloud.google.com/storage/

bprashanth committed:
Merge pull request #681 from aledbf/add-lb-status

Update Ingress status information in nginx controller

Commit:
Avoid unnecessary 'changed' for kubernetes-addons playbooks

Currently several addon spec file are always downloaded from Github,
each time Ansible run. Then installed to target configuration file after
inline changes.

This creates too many 'changed' noise and breaks the expectation that
Ansible run should be idempotent and only report 'changed' when really
does.

This change mute 'changed' message for downloading and inline replacing
part, and let the final install to /etc part report 'changed' or not.

eparis committed:
Merge pull request #701 from unicell/kube-addon-changed-when

Avoid unnecessary 'changed' for kubernetes-addons playbooks

gmarek committed:
GCSStable returns false in case of error

eparis committed:
Merge pull request #703 from gmarek/munger

GCSStable returns false in case of error

aledbf committed:
Allow traffic to default server _

eparis committed:
Merge pull request #666 from rutsky/coreos-check-vagrant-version

add requirement for minimal Vagrant version to use for Core OS images

bprashanth committed:
Readme for tls example.

bprashanth committed:
Merge pull request #704 from bprashanth/https_eg

Readme for tls example.

luxas committed:
Fixup the exechealthz code to minimize mistakes

joonathan committed:
Updated with correction for Ingress TLS spec

Fix for `error validating data: expected type array, for field spec.tls, got map; if you choose to ignore these errors, turn validation off with --validate=false`

bprashanth committed:
Merge pull request #705 from luxas/fix_docs

Fixup the exechealthz code to minimize mistakes

bprashanth committed:
Bump up glbc version.

bprashanth committed:
Merge pull request #652 from bprashanth/ing_resilience

Resilience fixes for the Ingress controller

thockin committed:
Use -o name on kubectl run

Also arrange tmux panes for easier keybaord input.

Also rename scripts to have meaning on splits.

rutsky committed:
use static IP for VMs on Virtualbox

At least K8s master can't have dynamic IP.

This also will help with master IP configuration, since it is known
in Vagrantfile now.

Fixes #696.

gitschaub committed:
Fixes ipv4 address resolution of etcd peers in etcd config

Updated, better syntax

rutsky committed:
add kube_apiserver_bind_address variable

rutsky committed:
set kube_apiserver_bind_address variable for Virtualbox

fixes #700

aledbf committed:
Wait until the pod it's running before getting information about it

stephenrlouie committed:
Created variables for flannel download url

eparis committed:
Merge pull request #654 from adamschaub/coreos_uninstall

uninstall CoreOS support: Touch up uninstall CoreOS wording. Added a missing check for is_coreos

rutsky committed:
add "etcd_interface" variable

Allows to specify host public interface on which etcd will listen.
Used to construct default advertisement addresses.

Useful on configuration when default interface is not connected to other
machines, such as in Vagrant+VirtualBox configuration.

aledbf committed:
Add git information during build

rutsky committed:
specify proper interface for etcd on Vagrant+VBox

eparis committed:
Merge pull request #655 from danehans/vag_ansible_ver

Updates Vagrant README

eparis committed:
Merge pull request #711 from adamschaub/etcd_template_fix

Fixes ipv4 address resolution of etcd peers in etcd config

bprashanth committed:
Merge pull request #706 from joonathan/patch-1

Updated with correction for Ingress TLS spec

bprashanth commented:
ping @devlinmr

spxtr committed:
Add a fake E2E tester.

eparis committed:
Merge pull request #714 from stephenrlouie/flannel-download-url

Created variable for flannel download url

eparis committed:
Merge pull request #715 from spxtr/fake-jenkins

Add a fake e2e tester.

eparis committed:
Merge pull request #724 from rutsky/etcd-interface

add "etcd_interface" variable

eparis committed:
Merge pull request #699 from eparis/fix-chip-link

submit queue: Update chip links

eparis committed:
Update angular and angular material

eparis committed:
submit-queue: Do not enforce release note labels on non-master

spxtr committed:
Submit queue: show the additions/deletions of PRs.

eparis committed:
Update pointer to release note process

eparis committed:
File rename for consistency

eparis committed:
Merge pull request #735 from spxtr/pr-size

Submit queue: show the additions/deletions of PRs.

eparis committed:
Display info to help understand the active queue order

eparis committed:
Merge pull request #736 from eparis/fix-release-note-label-on-non-master

Do not require (or set) release-note-* labels on non-master

eparis committed:
Minor layout change on the 'info' tab

eparis committed:
And update the whitelists while we are here

eparis committed:
expose the hueristic used to auto approve cherrypicks to other mungers

eparis committed:
Require labels on non-master PRs whose 'parents' do not have release-note-*

eparis committed:
Merge pull request #738 from eparis/release-note-non-master-hueristic

Require labels on non-master PRs whose 'parents' do not have release-note-*

eparis committed:
Merge pull request #740 from eparis/priority-info

submit-queue: Display info to help understand the active queue order

gmarek committed:
Make utils URL configurable

gmarek committed:
Add go1.6 to hack/verify-gofmt.sh and fail verify check if go version is unknown

eparis committed:
Fix the spacing on the info tab

eparis committed:
Remove wasted whitespace (especially nicer for mobile)

eparis committed:
Merge pull request #741 from gmarek/gofmt

Add go1.6 to hack/verify-gofmt.sh and fail verify check if go version…

eparis committed:
Merge pull request #742 from eparis/minor-display-on-info

submit-queue: Minor display changes to waste less space

thockin committed:
Merge pull request #709 from thockin/o-name

Use -o name on kubectl run

Commit:
Fixes pypy url by removing trailing slash

eparis committed:
Merge pull request #744 from danehans/pypy_fix

Fixes pypy url by removing trailing slash

eparis committed:
Merge pull request #737 from gmarek/bucket

Make utils URL configurable

stephenrlouie committed:
url_base naming convention to match flannel and kube

rutsky committed:
update K8s version used in github-release to 1.2.2

rutsky committed:
conditionally create /etc/hosts file

Now /etc/hosts file is not being touched (i.e. modification time is not
changed) if it already exists.

Removed unnecessary "changed" state in Ansible run summary.

eparis committed:
Merge pull request #748 from rutsky/touch-etc-hosts

conditionally create /etc/hosts file

rutsky committed:
make VBox static IPs not intersect with Flannel default range

In default configuration of flannel role 172.16.0.0/12 is used.

eparis committed:
Merge pull request #749 from rutsky/update-k8s-1.2.2

update K8s version used in github-release to 1.2.2

eparis committed:
Merge pull request #712 from rutsky/vagrant-static-ip

[Vagrant+VBox+Ansible] use static ip + explicitly set API Server bind address

stephenrlouie committed:
Added slash to kube url

eparis committed:
Merge pull request #750 from stephenrlouie/kube-slash

Added slash to kube url

spxtr committed:
Add job health to the internal e2e page.

eparis committed:
Merge pull request #729 from spxtr/e2e-health

Add job health to the internal e2e page.

eparis committed:
Merge pull request #747 from stephenrlouie/pypy_update

pypy url_base naming convention to match flannel and kube

gmarek committed:
Update munger tests so they will work for GCS based checker

bprashanth committed:
Merge pull request #707 from aledbf/default-server

Allow traffic to default server _ in nginx Ingress controller

gmarek committed:
Fix WeakStable in munger and add tests

eparis committed:
Merge pull request #751 from gmarek/e2e-test

Update munger tests so they will work for GCS based checker

eparis committed:
Merge pull request #754 from gmarek/weak

Fix WeakStable in munger and add tests

Commit:
Add default http block

Commit:
Add default http block

Commit:
Add default http block

Commit:
Add switches to enable Vault and handle ingresses without hosts

Commit:
Add switches to enable Vault and handle ingresses without hosts

Commit:
merge

bprashanth commented:
@devlinmr thats a crazy list of commits :) guessing it was a rebase error?

Commit:
Add client upload max

Commit:
Add facility to insert signer for vault SSL. Go vault lib ignores VAULT_SKIP_VERIFY

Commit:
Switch backends to https when ssl: true on frontend

Commit:
Fix defect when ingress has no labels

Commit:
Exclude binary

Commit:
Add debug

Commit:
Switch to using map instead of ingress object for greater control

Commit:
Tweak instantiation

Commit:
formatting errors on template

Commit:
Move checks into cascading if block

Commit:
Spacing

Commit:
tini not reqd to stop zombies

Commit:
1.6.2 - Add request_time

Commit:
Optimize- devlm/nginx-ingress:1.6.4

Commit:
Optimize- devlm/nginx-ingress:1.6.4

Commit:
Add TLS 1.0 1.1 1.2 nginx-ingress:1.6.5

Commit:
Add auto-renewal

test-foxish commented:
recomputing cla status...

Commit:
1.7.0- Add backend and key validation
